htdocs/core/class/commonobject.class.php

```diff
@@ -4125,8 +4125,16 @@ function insertExtraFields()
             $langs->load('admin');
             require_once DOL_DOCUMENT_ROOT.'/core/class/extrafields.class.php';
             $extrafields = new ExtraFields($this->db);
-            $extrafields->fetch_name_optionals_label($this->table_element);
-
+            $target_extrafields=$extrafields->fetch_name_optionals_label($this->table_element);
+            
+            //Eliminate copied source object extra_fiels that do not exist in target object
+            $new_array_options=array();
+            foreach ($this->array_options as $key => $value) {
+                if (in_array(substr($key,8), array_keys($target_extrafields)))
+                    $new_array_options[$key] = $value;
+            }
+            $this->array_options =$new_array_options;
```

Comment:
Can you change this to have

              foreach($new_array_options as $key => $value)		              foreach($this->array_options as $key => $value)

and remove the   $this->array_options =$new_array_options;

So function should works like you expect to, and when exiting function, entry properties were not modified (I hate to have a function that change the context outside of function) ?